### PR TITLE
Enable link check during documentation build and fix broken

### LINKED ISSUE #806: `tox -e docs`: linkcheck very flaky
This check is definitely useful and should be done regularly, but sadly is very flaky.

This is what I can think of:

* remove it from normal builds and make this check a part of the release, still having a regular check for dead links but having less flaky builds.
* wrap this step to be a proper test and introducing [pytest-rerunfailures](https://github.com/pytest-dev/pytest-rerunfailures) to the mix.
* run these checks every time but not fail, if dead links are detected (my least favorite appproach, because those tests tend to be ignored then)

### PR DESCRIPTION
**Pull request by gaborbernat**
- Fix links
- Enable link check during documentation build

Resolves #806.

### CODE REVIEW

File: docs/changelog.rst [hunk 1]
```diff
@@ -20,24 +20,24 @@ Features - 4.0.0a10
...
 - Add support for rewriting script invocations that have valid shebang lines when the ``TOX_LIMITED_SHEBANG`` environment
   variable is set and not empty - by :user:`gaborbernat`. (`#2208 <https://github.com/tox-dev/tox/issues/2208>`_)
-- Support for the ``--discover`` CLI flag - by :user:`gaborbernat`. (`#2245 <https://github.com/tox-dev/tox/issues/2245>`_)
+- Support for the ``--discover`` CLI flag - by :user:`gaborbernat`. (`#2245 <https://github.com/tox-dev/tox/pull/2245>`_)
 - Moved the python packaging logic into a dedicate package `pyproject-api <https://pypi.org/project/pyproject-api>`_ and
-  use it as a dependency - by :user:`gaborbernat`. (`#2274 <https://github.com/tox-dev/tox/issues/2274>`_)
-- Drop python 3.6 support - by :user:`gaborbernat`. (`#2275 <https://github.com/tox-dev/tox/issues/2275>`_)
+  use it as a dependency - by :user:`gaborbernat`. (`#2274 <https://github.com/tox-dev/tox/pull/2274>`_)
+- Drop python 3.6 support - by :user:`gaborbernat`. (`feature #2275 <https://github.com/tox-dev/tox/pull/2275>`_)
```
Comment (jugmac00) [inline docs/changelog.rst:26]: Where is `feature`coming from? And what it is for?

File: docs/changelog.rst [hunk 2]
```diff
@@ -92,22 +92,22 @@ Features - 4.0.0a8
 - Add support for ``{:}`` substitution in ini files as placeholder for the OS path separator - by :user:`gaborbernat`. (`#2125 <https://github.com/tox-dev/tox/issues/2125>`_)
 - When cleaning directories (for tox environment, ``env_log_dir``, ``env_tmp_dir`` and packaging metadata folders) do not
   delete the directory itself and recreate, but instead just delete its content (this allows the user to cd into it and
-  still be in a valid folder after a new run) - by :user:`gaborbernat`. (`#2139 <https://github.com/tox-dev/tox/issues/2139>`_)
+  still be in a valid folder after a new run) - by :user:`gaborbernat`. (`#2130 <https://github.com/tox-dev/tox/issues/2130>`_)
```
Comment (jugmac00) [inline docs/changelog.rst:95]: This looks like the wrong number. #2139 looks right for me.

File: docs/changelog.rst [hunk 3]
```diff
@@ -131,10 +131,10 @@ Features - 4.0.0a7
...
 - Inline plugin support via ``tox_.py``. This is loaded where the tox config source is discovered. It's a Python file
   that can contain arbitrary Python code, such as definition of a plugin. Eventually we'll add a plugin that allows
-  succinct declaration/generation of new tox environments - by :user:`gaborbernat`. (`#1963 <https://github.com/tox-dev/tox/issues/1963>`_)
+  succinct declaration/generation of new tox environments - by :user:`gaborbernat`. (`#1963 <https://github.com/tox-dev/tox/pull/1963>`_)
 - Introduce the installer concept, and collect pip installation into a ``pip`` package, also attach to this
-  the requirements file parsing which got a major rework - by :user:`gaborbernat`. (`#1991 <https://github.com/tox-dev/tox/issues/1991>`_)
-- Support CPython ``3.10`` -by :user:`gaborbernat`. (`#2014 <https://github.com/tox-dev/tox/issues/2014>`_)
+  the requirements file parsing which got a major rework - by :user:`gaborbernat`. (`feature #1991 <https://github.com/tox-dev/tox/pull/1991>`_)
```
Comment (jugmac00) [inline docs/changelog.rst:136]: One more addition of the word `feature`.

File: docs/changelog.rst [hunk 4]
```diff
@@ -152,23 +152,23 @@ Bugfixes - 4.0.0a7
   without the ``testenv:`` prefix - by :user:`jugmac00`, :user:`obestwalter`, :user:`eumiro`. (`#1985 <https://github.com/tox-dev/tox/issues/1985>`_)
 - Fix legacy list env command for empty/missing envlist - by :user:`jugmac00`. (`#1987 <https://github.com/tox-dev/tox/issues/1987>`_)
 - Requirements and constraints files handling got reimplemented, which should fix all open issues related to this area
-  - by :user:`gaborbernat`. (`#1991 <https://github.com/tox-dev/tox/issues/1991>`_)
+  - by :user:`gaborbernat`. (`bugfix #1991 <https://github.com/tox-dev/tox/pull/1991>`_)
```
Comment (jugmac00) [inline docs/changelog.rst:155]: I do not see a pattern when the classifiers are added and when not.

File: docs/changelog.rst [hunk 5]
```diff
@@ -152,23 +152,23 @@ Bugfixes - 4.0.0a7
...
+- Move from ``appdirs`` to ``platformdirs`` - by :user:`gaborbernat`. (`#2117 <https://github.com/tox-dev/tox/pull/2117>`_)
+- Move from ``toml`` to ``tomli`` - by :user:`gaborbernat`. (`#2118 <https://github.com/tox-dev/tox/pull/2118>`_)
 
 Improved Documentation - 4.0.0a7
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 - Start documenting the plugin interface. Added :meth:`tox_register_tox_env <tox.plugin.spec.tox_register_tox_env>`,
   :meth:`tox_add_option <tox.plugin.spec.tox_add_option>`,
   :meth:`tox_add_core_config <tox.plugin.spec.tox_add_core_config>`,
-  ``tox_configure`` - by :user:`gaborbernat`. (`#1991 <https://github.com/tox-dev/tox/issues/1991>`_)
+  ``tox_configure`` - by :user:`gaborbernat`. (`documentation #1991 <https://github.com/tox-dev/tox/pull/1991>`_)
```
Comment (jugmac00) [inline docs/changelog.rst:169]: Why add `documentation` here when the heading reads `Improved documentation`? Are those keywords added by the tool or manually?

**Review by jugmac00 — COMMENTED** [review]
I am -1 for using the check in the default `docs` env. Building docs just took 89 seconds instead of 7 seconds.

Also, I am not sure what was manually changed in this PR and what the tool did.

**Comment by gaborbernat** [comment]
> I am -1 for using the check in the default `docs` env. Building docs just took 89 seconds instead of 7 seconds.

You can always bail it out by running it with ``tox -e docs --- ``

> Also, I am not sure what was manually changed in this PR and what the tool did.

Everything was manual/

File: docs/changelog.rst (hunk 1 above)
Comment (gaborbernat) [reply docs/changelog.rst:26]: When you add two such hyperlinks (PRs adding both two or more of bugfix, doc, feature at the same time) sphinx complains that you defined two references with the same key, this differentiates it.

File: docs/changelog.rst (hunk 3 above)
Comment (gaborbernat) [reply docs/changelog.rst:136]: Same as above.

File: docs/changelog.rst (hunk 4 above)
Comment (gaborbernat) [reply docs/changelog.rst:155]: See above.

File: docs/changelog.rst (hunk 5 above)
Comment (gaborbernat) [reply docs/changelog.rst:169]: Need something to avoid duplicate reference error.

File: docs/changelog.rst (hunk 2 above)
Comment (gaborbernat) [reply docs/changelog.rst:95]: This observation here is correct.

Comment (gaborbernat) [reply docs/changelog.rst:95]: Seems already fixed on rewrite branch.

**Pull request merged**